lib/importproject.cpp

```diff
@@ -167,24 +167,29 @@ void ImportProject::FileSettings::setIncludePaths(const std::string &basepath, c
     includePaths.swap(I);
 }
 
-void ImportProject::import(const std::string &filename)
+ImportProject::Type ImportProject::import(const std::string &filename)
 {
     std::ifstream fin(filename);
     if (!fin.is_open())
-        return;
-    if (filename.find("compile_commands.json") != std::string::npos) {
+        return MISSING;
+    if (endsWith(filename, "compile_commands.json", 22)) {
```

Comment:
it's much better to use `endsWith`! Thank you. I refactored the function so the `endlen` doesn't have to be manually written. Please update your patch.